src/plugins/data/common/es_query/es_query/from_kuery.ts

```diff
@@ -35,22 +35,20 @@ export function buildQueryFromKuery(
 }
 
 function buildQuery(
-  indexPattern: IIndexPattern | null,
+  indexPattern: IIndexPattern | undefined,
   queryASTs: KueryNode[],
   config: Record<string, any> = {}
 ) {
-  const compoundQueryAST: KueryNode = nodeTypes.function.buildNode('and', queryASTs);
-  const kueryQuery: Record<string, any> = toElasticsearchQuery(
-    compoundQueryAST,
-    indexPattern,
-    config
-  );
+  const compoundQueryAST = nodeTypes.function.buildNode('and', queryASTs);
+  const kueryQuery = toElasticsearchQuery(compoundQueryAST, indexPattern, config);
 
-  return {
-    must: [],
-    filter: [],
-    should: [],
-    must_not: [],
-    ...kueryQuery.bool,
-  };
+  return Object.assign(
+    {
+      must: [],
+      filter: [],
+      should: [],
+      must_not: [],
+    },
+    kueryQuery.bool
+  );
```

Comment:
Curious; why `Object.assign()` here instead of spread syntax?

No need to change it back -- I don't really care either way -- I'm just interested to know if there was a technical reason for it, or if it's just a stylistic preference.